source/gameengine/Launcher/LA_Launcher.cpp

```diff
@@ -287,9 +288,11 @@ void LA_Launcher::ExitEngine()
    * which allows the scene to safely delete them :)
    * see: (space.c)->start_game
    */
-
   PyDict_Clear(PyModule_GetDict(m_gameLogic));
 
+  PyDict_Clear(m_profileDict);
+  PyDict_Clear(m_globalDict);
```

Comment:
The globalDict must only be freed if exit request is to quit game, not if you start another .blend.

I propose this diff with your changes + mine: https://pastebin.com/6MpmxRXi

Reply:
> Also I'd add m_profileDict = nullptr; after PyDict_Clear(m_profileDict)
I was just trying to follow what the current code does. So, should I reset other global members like `m_profileDict` in the same way?